dwave/system/samplers/clique.py

```diff
@@ -0,0 +1,186 @@
+        try:
+            return self._parameters
+        except AttributeError:
+            pass
+
+        self._parameters = parameters = self.child.parameters.copy()
+
+        # this sampler handles scaling
+        parameters.pop('auto_scale', None)
+        parameters.pop('bias_range', None)
+        parameters.pop('quadratic_range', None)
+
+        return parameters
+
+    @property
+    def properties(self):
+        try:
+            return self._properties
+        except AttributeError:
+            pass
+
+        self._properties = dict(qpu_properties=self.child.properties)
+        return self.properties
+
+    @property
+    def largest_clique_size(self):
+        """The maximum number of variables."""
+        return len(self.largest_clique())
+
+    def largest_clique(self):
+        """Return a largest-size clique embedding."""
+        return busgraph_cache(self.target_graph).largest_clique()
+
+    def sample(self, bqm, chain_strength=None, **kwargs):
+        """Sample from the specified binary quadratic model.
+
+        Args:
+            bqm (:class:`~dimod.BinaryQuadraticModel`):
+                Any binary quadratic model with up to
+                :attr:`.largest_clique_size` variables. This BQM is embedded
+                using a dense clique embedding.
+
+            chain_strength (float, optional):
+                The (relative) chain strength to use in the embedding. By
+                default a chain strength of `1.5sqrt(N)` where `N` is the size
+                of the largest clique, as returned by
+                :attr:`.largest_clique_size`.
+
+            **kwargs:
+                Optional keyword arguments for the sampling method, specified
+                per solver in :attr:`.DWaveCliqueSampler.parameters`.
+                D-Wave System Documentation's
+                `solver guide <https://docs.dwavesys.com/docs/latest/doc_solver_ref.html>`_
+                describes the parameters and properties supported on the D-Wave
+                system. Note that `auto_scale` is not supported by this
+                sampler, because it scales the problem as part of the embedding
+                process.
+
+        """
+
+        # some arguments should not be overwritten
+        if 'auto_scale' in kwargs:
+            raise TypeError("sample() got an unexpected keyword argument "
+                            "'auto_scale'")
+        if 'bias_range' in kwargs:
+            raise TypeError("sample() got an unexpected keyword argument "
+                            "'bias_range'")
+        if 'quadratic_range' in kwargs:
+            raise TypeError("sample() got an unexpected keyword argument "
+                            "'quadratic_range'")
+
+        # handle circular import. todo: fix
+        from dwave.system.composites.embedding import FixedEmbeddingComposite
+
+        # get the embedding
+        embedding = find_clique_embedding(bqm.variables, self.target_graph,
```

Comment:
Note, this method will return an empty embedding if the maximum size is exceeded -- might be good to check and toss an error where appropriate.

Reply:
Good point, and good test to add.

Reply:
It actually currently throws `ValueError: no embedding found` which isn't too bad (yay abstraction handling it!). But I something more explicit would be good.